tests/integration/ambient/cacert_rotation_test.go

```diff
@@ -64,13 +64,13 @@ func TestIntermediateCertificateRefresh(t *testing.T) {
 
 			// Update CA with new intermediate cert
 			if err := cert.CreateCustomCASecret(t,
-				"ca-cert-alt-2.pem", "ca-key-alt-2.pem",
-				"cert-chain-alt-2.pem", "root-cert-alt.pem"); err != nil {
+				"ca-cert.pem", "ca-key.pem",
+				"cert-chain.pem", "root-cert.pem"); err != nil {
 				t.Errorf("failed to update CA secret: %v", err)
 			}
 
 			// perform one retry to handle race condition where ztunnel cert is refreshed before Istiod certificates are reloaded
-			retry.UntilSuccess(func() error {
+			retry.UntilSuccessOrFail(t, func() error {
```

Comment:
this wasn't asserting, so they test always passed even though it was broken. Then it left things in a permanently broken state